torchvision/ops/misc.py

```diff
@@ -253,3 +253,47 @@ def _scale(self, input: Tensor) -> Tensor:
     def forward(self, input: Tensor) -> Tensor:
         scale = self._scale(input)
         return scale * input
+
+
+class MLP(torch.nn.Sequential):
+    """This block implements the multi-layer perceptron (MLP) module.
+
+    Args:
+        in_channels (int): Number of channels of the input
+        hidden_channels (List[int]): List of the hidden channel dimensions
+        norm_layer (Callable[..., torch.nn.Module], optional): Norm layer that will be stacked on top of the convolution layer. If ``None`` this layer wont be used. Default: ``None``
+        activation_layer (Callable[..., torch.nn.Module], optional): Activation function which will be stacked on top of the normalization layer (if not None), otherwise on top of the conv layer. If ``None`` this layer wont be used. Default: ``torch.nn.ReLU``
+        inplace (bool): Parameter for the activation layer, which can optionally do the operation in-place. Default ``True``
+        bias (bool): Whether to use bias in the linear layer. Default ``True``
+        dropout (float): The probability for the dropout layer. Default: 0.0
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        hidden_channels: List[int],
+        norm_layer: Optional[Callable[..., torch.nn.Module]] = None,
+        activation_layer: Optional[Callable[..., torch.nn.Module]] = torch.nn.ReLU,
+        inplace: Optional[bool] = True,
+        bias: bool = True,
+        dropout: float = 0.0,
```

Comment:
I'm slightly confused over this choice. 

We should rather not append dropout layer if dropout is 0.0. 

Maybe 
```suggestion
        dropout: Optional[float] = None,
```

And check if dropout is not None and then add the layer?

Reply:
That was originally the plan but it would make it very hard to maintain BC for MLPBlock. See ViT's usage for more details. It was possible but would require a lot of ugly code. Leaving the dropout layer should have no performance issues.

Reply:
While Dropout is not going to cause any performance issues. But I'm afraid we aren't encouraging best practices. 

I thought we could just leave MLP in ViT as it as (not inherit from ops) And make the MLP layer a bit better as suggested. 

Again it's either way, both have pros and cons.

Reply:
Would it be an option to implement this as it was done in the [multimodal package](https://github.com/facebookresearch/multimodal/blob/cef848f3d0313d4a682f2b10ca7f8506da820451/torchmultimodal/modules/layers/mlp.py#L59)? 
```
if dropout > 0:
    layers.append(nn.Dropout(dropout))
```